src/test/java/com/google/sps/ImageSelectionTest.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.image;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+@RunWith(JUnit4.class)
+public final class ImageSelectionTest {
+
+  private final Set<String> keywords = new LinkedHashSet<>();
+
+  // This is how every Bing image link starts
+  private static final String EXPECTED_LINK = "https://tse";
+
+  private void checkScrapingOutputLink() {
```

Comment:
This test case won't be executed─need @Test annotation.

(Note that with in true TDD fashion, you'd catch this mistake because you start with a failing test.)

Reply:
checkScrapingOutputLink() is not a test. Each test has the same final checking steps, so the purpose of this function is to remove duplicate code.

Reply:
Ooops, you're right! See my other comment though, once you write this custom assertion like the following

`private void assertXyz(Set<String> keywords)`

it will become obvious right away (most assertions start with "assert" and there are no test methods that take arguments)